weto/src/main/java/fi/uta/cs/weto/util/NotificationManager.java

```diff
@@ -39,14 +37,225 @@ public void contextInitialized(ServletContextEvent event) {
 
         int notificationEmailInterval = Integer.parseInt(
                 WetoUtilities.getPackageResource("notification.emailInterval.minutes"));
+        int permissionExpirationCheckInterval = Integer.parseInt(
+                WetoUtilities.getPackageResource("notification.permissionExpirationCheckInterval.hours"));
         scheduler.scheduleAtFixedRate(new NotificationEmailTask(), 1, notificationEmailInterval, TimeUnit.MINUTES);
+
+        scheduler.scheduleAtFixedRate(new DeadlineNotificationTask(), 1, permissionExpirationCheckInterval, TimeUnit.MINUTES);
     }
 
     @Override
     public void contextDestroyed(ServletContextEvent event) {
         scheduler.shutdownNow();
     }
 
+
+
+
+    private class DeadlineNotificationTask implements Runnable {
+
+        @Override
+        public void run() {
+
+            ArrayList<Integer> triggeringPermissions = new ArrayList<>();
+            triggeringPermissions.add(PermissionType.SUBMISSION.getValue());
+            triggeringPermissions.add(PermissionType.GRADING.getValue());
+            triggeringPermissions.add(PermissionType.RESULTS.getValue());
+            triggeringPermissions.add(PermissionType.GRADE_CHALLENGE.getValue());
+
+            Connection masterCon = connectionManager.getConnection("master");
+            ArrayList<Permission> activeTasks = new ArrayList<>();
+
+            try {
+                activeTasks = Permission.selectActive(masterCon);
+                logger.debug("loaded " + activeTasks.size() + " active courses");
+            } catch (Exception e) {
+                logger.error(e);

```

Comment:
Add message to error logging
```suggestion
                logger.error("Failed to load active courses", e);
```